### PR TITLE
Becca Hyland

### PR DESCRIPTION
**Pull request by BeccaHyland**
Through Iteration 4

### CODE REVIEW

File: lib/market.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    @vendors.map do |vendor|
+      vendor.name
+    end
+  end
+
+  def vendors_that_sell(item)
+    @vendors.find_all do |vendor|
+      vendor.inventory.has_key?(item)
+    end
+  end
```
Comment (s-espinosa) [inline lib/market.rb]: 👍 Like this!

File: lib/market.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  def sorted_item_list
+    items = []
+    @vendors.each do |vendor|
+      vendor.inventory.each do |item, amount|
+        items << item
+      end
+    end
+    items.uniq.sort
+  end
```
Comment (s-espinosa) [inline lib/market.rb]: Like this pattern! Could use a Ruby enumerable, but really like that you're able to get `each` to do what you need it to.

File: lib/market.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  def total_inventory
+    total = Hash.new(0)
+    @vendors.each do |vendor|
+      vendor.inventory.each do |item, amount|
+        total[item] += amount
+      end
+    end
+    total
+  end
```
Comment (s-espinosa) [inline lib/market.rb]: 👍

File: lib/market.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+  def sell(item, amount)
+    unless !sellable?(item, amount)
```
Comment (s-espinosa) [inline lib/market.rb]: Consider an `if` here?

```
if sellable?(item, amount)
```

File: lib/market.rb [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      amount_to_sell = amount
+      until amount_to_sell == 0
```
Comment (s-espinosa) [inline lib/market.rb]: Like this approach.

File: lib/market.rb [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+  def get_vendor_with_item(item)
+    @vendors.find do |vendor|
+      vendor.check_stock(item) != 0
+    end
+  end
+
+  def sellable?(item, amount)
+    total_inventory[item] >= amount
+  end
```
Comment (s-espinosa) [inline lib/market.rb]: 👍

File: test/market_test.rb [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+  end
+
+  def test_it_can_sell_item_if_item_in_stock_for_a_vendor
+    @m.add_vendor(@vendor_1)
+    @m.add_vendor(@vendor_2)
+    @m.add_vendor(@vendor_3)
+
+    refute @m.sell("Peaches", 200)
+    refute @m.sell("Onions", 1)
+    assert @m.sell("Banana Nice Cream", 5)
```
Comment (s-espinosa) [inline test/market_test.rb]: Like the edge case testing here!
